File: scratch/inject_modal.js
const fs = require('fs');

try {
  const indexContent = fs.readFileSync('index.html', 'utf8');
  const startTag = '<!-- Contact Modal (FormSubmit) -->';
  const endTag = '<!-- Legal Modal -->';
  
  const startIdx = indexContent.indexOf(startTag);
  const endIdx = indexContent.indexOf(endTag);
  
  if (startIdx !== -1 && endIdx !== -1) {
    const modalHtml = indexContent.substring(startIdx, endIdx);
    console.log('Modal extraído con éxito (largo:', modalHtml.length, ')');
    
    const files = [
      'calendario.html',
      'carta-del-dia.html',
      'fase-lunar.html',
      'grimorio.html',
      'horoscopo.html',
      'numerologia.html',
      'significado-suenos.html',
      'tienda.html'
    ];
    
    files.forEach(file => {
      if (fs.existsSync(file)) {
        let content = fs.readFileSync(file, 'utf8');
        if (!content.includes('id="contact-modal"')) {
          if (content.includes('<!-- Legal Modal -->')) {
            content = content.replace('<!-- Legal Modal -->', modalHtml + '\n\n  <!-- Legal Modal -->');
          } else {
            content = content.replace('</body>', modalHtml + '\n</body>');
          }
          fs.writeFileSync(file, content, 'utf8');
          console.log(`Inyectado modal en ${file}`);
        } else {
          console.log(`${file} ya tiene el modal.`);
        }
      }
    });
  } else {
    console.log('No se pudo encontrar el bloque delimitado en index.html');
  }
} catch (e) {
  console.error(e);
}

File: scratch/inject_modal.py
import re
import os

with open('index.html', 'r', encoding='utf-8') as f:
    index_content = f.read()

# Extraer el bloque del modal de contacto
modal_match = re.search(r'(<!-- Contact Modal.*?FormSubmit -->.*?</div>\s*</div>\s*</div>)', index_content, re.DOTALL)
if modal_match:
    modal_html = modal_match.group(1)
    print("Modal extraído correctamente")
    
    files = ['calendario.html', 'carta-del-dia.html', 'fase-lunar.html', 'grimorio.html', 'horoscopo.html', 'numerologia.html', 'significado-suenos.html', 'tienda.html']
    for file in files:
        if os.path.exists(file):
            with open(file, 'r', encoding='utf-8') as f:
                content = f.read()
            if 'id="contact-modal"' not in content:
                # Inyectar el bloque antes de <!-- Legal Modal --> o si no antes de </body>
                if '<!-- Legal Modal -->' in content:
                    new_content = content.replace('<!-- Legal Modal -->', modal_html + '\n\n  <!-- Legal Modal -->')
                else:
                    new_content = content.replace('</body>', modal_html + '\n</body>')
                with open(file, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Inyectado modal de contacto en {file}")
            else:
                print(f"{file} ya contiene el modal")
else:
    print("Error: No se pudo extraer el modal de index.html")
